### PR TITLE
Sort media sources

### LINKED ISSUE #6955: Media Browser Panel - Choose Player dialog not sorted alphabetically (0.115.0b7)
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem

Was gonna label this as a FR but it feels more like a bug tbh... The dialog for "Choose Player" should be sorted alphabetically IMO. If I know I want to play something on my Bedroom speaker, I would expect that near the top, not to have to go hunting for it. 

This is kind of messy imho (and there's people with a lot more media players than me):

image

## Expected behavior

Alphabetical

## Steps to reproduce

1. Open media browser panel, look at sorting...

## Environment

- Home Assistant release with the issue: 0.115.0b7
- Last working Home Assistant release (if known): n/a
- Browser and browser version: Chrome
- Operating system: macOS

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

### PR DESCRIPTION
**Pull request by ludeeus**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: closes #6955
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by bramkragten — APPROVED** [review]

File: src/panels/media-browser/hui-dialog-select-media-player.ts [hunk 1]
```diff
@@ -55,13 +55,15 @@ export class HuiDialogSelectMediaPlayer extends LitElement {
...
           >
-          ${this._params.mediaSources.map(
-            (source) => html`
-              <paper-item .itemName=${source.entity_id}
-                >${source.attributes.friendly_name}</paper-item
-              >
-            `
-          )}
+          ${this._params.mediaSources
+            .sort()
```
Comment (bramkragten) [inline src/panels/media-browser/hui-dialog-select-media-player.ts:59]: You need to sort by name right? So need a sort function that compares the name

Comment (bramkragten) [inline src/panels/media-browser/hui-dialog-select-media-player.ts:59]:
```suggestion
            .sort((a, b) =>
          compare(a.attributes.friendly_name || "", b.attributes.friendly_name || ""))
```

Comment (ludeeus) [reply src/panels/media-browser/hui-dialog-select-media-player.ts:59]: Yes?

File: src/panels/media-browser/hui-dialog-select-media-player.ts [hunk 2]
```diff
@@ -56,7 +56,8 @@ export class HuiDialogSelectMediaPlayer extends LitElement {
             )}</paper-item
           >
           ${this._params.mediaSources
-            .sort()
+            .sort((a, b) =>
+          compare(a.attributes.friendly_name || "", b.attributes.friendly_name || ""))
```
Comment (bramkragten) [inline src/panels/media-browser/hui-dialog-select-media-player.ts]:
```suggestion
          compare(computeStateName(a), computeStateName(b)))
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**